# Problem 1:

If you are able to read this, then the steps folllowed were correct. First, I have entered the Accelerated_Intro_WilkinsonExams file cloned into my computer (VM) through the terminal, and executed the command **jupyter notebook** to open the folder. Then just created a new notebook with a bash kernel and renamed it to Exam_1_Answers.


# Problem 2:

First, I use the command *mkdir* to create the directory and we assign it the name Exam_1. This will create a new directory in the folder I am inside of(Accelerated_Intro_WIlkinsonExams). After that, by means of the *ls* "listing" command I can see the contents of that folder, including the newly made Exam_1 directory. In order to see the size, permission and ownership of each file, we put the flags *-l* (list type) and *-h* (human readable) to the command *ls*. 
Te output is a list of nearly all files (*-f* for full list) with the **drwx** code for permissions on the first column, the two next columns (osboxes osboxes) make reference to the group and user. Next to those, the size of the folder is given in kilo and Megabytes (or simply bytes). Last part is the date the file was created.
In the case of the Exam_1 directory, the permissions can be read as:
d > a directory
r > readable
w > permission to write
x > permission to execute
Every three letters refer to the permissions conceaded to User, Group, and everybody respectivelly. For this directory then, all users and groups can read, write and execute the directory, but a non-related individual in the system (everyone) can only read and execute.
Finally, I removed the directory with a *rm* remove dir command and a *-r* flag to make sure is deleted.


In [4]:
mkdir Exam_1
ls -lh 
rm -r Exam_1
ls -lh

total 2.4M
-rw-rw-r-- 1 osboxes osboxes 1.2M Sep  3 10:38  conv.txt
-rw-rw-r-- 1 osboxes osboxes  147 Sep  3 10:38  DISCLAIMER
drwxrwxr-x 2 osboxes osboxes 4.0K Sep  5 07:33  Exam_1
-rw-rw-r-- 1 osboxes osboxes  961 Sep  5 07:23  Exam_1_Answers.ipynb
-rw-rw-r-- 1 osboxes osboxes  896 Sep  3 11:30 'Exam answers preparacion.ipynb'
-rw-rw-r-- 1 osboxes osboxes 5.5K Sep  3 10:38 'Exam Week 1.ipynb'
-rw-rw-r-- 1 osboxes osboxes 3.9K Sep  3 10:38 'Exam Week 2.ipynb'
-rw-rw-r-- 1 osboxes osboxes 4.2K Sep  3 10:38  Germplasm.tsv
-rw-rw-r-- 1 osboxes osboxes  141 Sep  3 10:38  how_to_convert.txt
-rw-rw-r-- 1 osboxes osboxes 1.1K Sep  3 10:38  LICENSE
-rw-rw-r-- 1 osboxes osboxes  637 Sep  3 10:38  LocusGene.tsv
-rw-rw-r-- 1 osboxes osboxes 1.2M Sep  3 10:38  Locus_Germplasm_Phenotype_20130122.txt
-rw-rw-r-- 1 osboxes osboxes  743 Sep  3 10:38  README.md
total 2.4M
-rw-rw-r-- 1 osboxes osboxes 1.2M Sep  3 10:38  conv.txt
-rw-rw-r-- 1 osboxes osboxes  147 Sep  3 10:38  DISCLAIMER
-rw-rw-r-- 1 osb

# Problem 3

In order to only show the header of the file, I have employed a *head* command wich displays the lines of the file from the top to the bottom. IN this case, as I only need to display the first line, *head* is followed by a -1 which refers to the number of lines to be shown.

In [5]:
head -1 Locus_Germplasm_Phenotype_20130122.txt

Locus_name)	Germplasm_name	phenotype	pubmed_id


# Problem 4

In order to know the total lines of a file, I have used the command *wc* (word count). This command can be used on its own, but I have piped it to the outcome of a *cat* command just to add some interest. Although the command is called word count it can also be used to ask for the number of lines with the proper flag.
*wc -l* is asking for just the number of lines (flag *-l*)

In [6]:
cat Locus_Germplasm_Phenotype_20130122.txt|wc -l

7216


# Problem 5

The simplest way of doing this task if the size of the file is known, as it is the case, is to use a *tail* command with the total number of lines minus 1. *Tail* is the opposite of *Head*, as it displays the selected number of lines from the file starting from the bottom and towards the top. The output of the *tail* command is therefore the total of lines except for the first one, and its stored in a file named Data_Only.csv as required. In order to prove that the file is indeed all lines minus header, the total number of lines are counted in both files with the *wc -l* command. The original file had 7216 lines, and the Data_Only.csv file has exactly 7216-1 = 7215.

In [33]:
tail -7215 Locus_Germplasm_Phenotype_20130122.txt> Data_Only.csv
wc -l Locus_Germplasm_Phenotype_20130122.txt
wc -l Data_Only.csv
rm Data_Only.csv

7216 Locus_Germplasm_Phenotype_20130122.txt
7215 Data_Only.csv


My second attept of doing this in a more sophisticated way didn't go as planned, and I still cannot find the error or why does the BABA file only have 7112 lines instead of 7215. First, I displayed the two strating lines of the file Locus_Germplasm... so I can confirm if the sustraction of the header is done correctly. A *grep* searching for one of the words in the header (pubmed_id) displays only one result, the one that is on the header and takes the whole header line as shown by the *cat*. A second *grep* exactly like the first one but with the *-v* flag was supposed to take all lines of the file minus the header and store them in a BABA named file. But the *wc* command shows that instead of the expected 7215 the BABA file has 7112. I still don't really know where the missing lines went and why.

In [5]:
head -2 Locus_Germplasm_Phenotype_20130122.txt
grep -w "pubmed_id" Locus_Germplasm_Phenotype_20130122.txt| cat
grep -w -v "pubmed_id" Locus_Germplasm_Phenotype_20130122.txt > BABA
head -1 BABA
wc -l BABA
rm BABA


Locus_name)	Germplasm_name	phenotype	pubmed_id
AT1G01040	CS3828	Increased abundance of miRNA precursors.	17369351
Locus_name)	Germplasm_name	phenotype	pubmed_id
AT1G01040	CS3828	Increased abundance of miRNA precursors.	17369351
7112 BABA


# Problem 6

As the word mentioned is "root", therefore all variations or plural of the mentioned word will be ignored. To show all the lines with the word "root" on their phenotype column (which is the only column that displays text in the file, and made this much easier) I just had to do a *grep* command with the *-i* flag to avoid case specificness, *-n* to show the line number and *-w* to search for whole words. 

In [74]:
grep -inw "root" Locus_Germplasm_Phenotype_20130122.txt

23:AT1G01550	bps1-2	Growth of bps1-2 mutants on CPTA-supplemented medium resulted in partial rescue of both leaf and root defects. bps1-2 mutants grown on control growth medium have small radialized leaves with very little vascular tissue, and very short misshapen knotted-looking roots. By contrast, bps1-2 mutants grown on CPTA-supplemented medium produced larger flattened leaves that contained primary and secondary veins, and smooth elongated roots.	17217459
24:AT1G01550	bps1-2	Mutant roots were abnormal; primary and lateral roots were short, root hairs formed close to the root apex, and root defects were most severe when plants were grown at low temperature.	15458645
26:AT1G01550	bps1-2	Under high light conditions (approximately 200 &mu;E.m-2.sec-1), CPTA-treated seedlings were completely photobleached and mutants showed rescue of both leaf and root development. Under moderate (approximately 100 &mu;E.m-2.sec-1) and low (30 &mu;E.m-2.sec-1) light intensities, seedlings are incomplete

253:AT1G08260	tilted1-4	Twenty-five percent of embryos show abnormal cell divisions during embryogenesis-large cells- decreased rate of cell division. Division of the hypohyseal cell abnormal leading to abnormal placement of the root pole. Homozygotes show delayed flowering, abnormal floraly phyllotaxy and abnormal ovules.	16278345
255:AT1G08370	dcp1-2	The homozygous progeny is seedling lethal, showed arrested postembryonic development including cotyledon expansion, development of vascular networks, root elongation, and shoot development.	17485080
257:AT1G08430	Atalm1-KO	The mutant exhibits very low levels of root  growth and malate release in the presence of  Al (0.2 nmol plant-1 24 h-1).
289:AT1G09100	rpt5b	Insensitive to high-glucose repression of cotyledon expansion, chlorophyll accumulation, true-leaf development and root elongation. The response is specific to glucose but not to osmotic changes.	17081979
294:AT1G09110	rpt5b	Insensitive to high-glucose repression of cotyledon expa

585:AT1G19220	CS24629	The phenotype of the double mutant is most obvious at its seedling stage, with its most prominent phenotype being severely impaired  lateral root formation. Its primary roots fail to produce lateral roots in 2-week-old seedlings. However, double mutant seedlings start to generate several lateral roots after ~2 weeks of growth, and their morphological appearance is normal.	15659631
589:AT1G19220	CS24630	The phenotype of the double mutant is most obvious at its seedling stage, with its most prominent phenotype being severely impaired  lateral root formation. Its primary roots fail to produce lateral roots in 2-week-old seedlings. However, double mutant seedlings start to generate several lateral roots after ~2 weeks of growth, and their morphological appearance is normal.	15659631
592:AT1G19220	arf7-201 arf19-101	Dark grown seedlings on sirtinol have both long primary roots and long hypocotyls.  More resistant to 2,4-D and IAA than single mutants in both light and d

882:AT1G27320	stip-d ahk2-2 ahk3-3 cre1-12	Small SAM and limited root growth.	20110319
894:AT1G27760	atsat32	Reduced fertility. Increased root length (longer roots). Hypersensitive to salt stress.	19210750
906:AT1G28560	srd2	temperature-sensitive mutant. srd2 seedlings grown at  the permissive temperature (22C) appeared almost normal. At 28C,  both hypocotyl and root explants of the srd2 mutant could  not establish shoot apical meristems de novo. When  primary root segments of srd2 were cultured on root-inducing  medium (RIM) at 28C, abnormal lateral root primordia  lacking functional apical meristems were frequently observed.  except that they had fewer lateral roots than the wild-type  seedlings. At the restrictive temperature (28C however, the development of true leaves and primary roots in srd2  seedlings was severely interrupted. Whereas root explants of the srd2 mutant formed  calli on callus-inducing medium (CIM), at either 22Cor 28C,  callus formation from srd2 hypocotyl explan

1163:AT1G51190	plt3-1/plt2-2/plt1-4	Rootless.Embryonic root differentiates 3 days after germination. Adventitious roots arrest 6 days after germination.	17960244
1199:AT1G52340	aba2-2	Mutant showed a similar sensitivity to 25 mM trehalose as wild type (inhibition of root growth).	17031512
1237:AT1G53700	wag1-1/wag2-1	Morphologically similar to wild type. When seedlings were grown on vertical plates, root exhibited pronounced wavy pattern.	16460509
1238:AT1G53700	wag1-2	Morphologically similar to wild type. When seedlings were grown on vertical plates, root exhibited slight wavy pattern. Overall phenotype identical to wag1-1 mutant allele.	16460509
1239:AT1G53700	wag1-2/wag2-1	Morphologically similar to double mutant wag1-1/wag2-1. When seedlings were grown on vertical plates, root exhibited pronounced wavy pattern. Gravitropism was not affected in mutant plants. Double mutant plants were more resistant to inhibition of root curling by auxin transport inhibitor NPA than wild type plants

1402:AT1G64440	reb1-1	Decreased root length (67% of that of wildtype) when grown on D-galactose-free medium.	12419184
1403:AT1G64440	reb1-1	Less galactose in root cell wall compared to wildtype. No difference in shoot cell wall monosaccharide composition.	16500990
1405:AT1G64440	reb1-1	Root growth significantly diminished.	
1406:AT1G64440	reb1-1	Root length defect was restored when mutant was grown on 10mM galactose.	16500990
1413:AT1G64520	rpn12a-1 (C24)	The rpn12a-1 mutant produces both the WT RPN12a protein and a RPN12a(1-186):ARP(exon2):NPTII fusion protein. The mutant form of the protein can coelute with subunits of the 26S proteasome from the regulatory and core particles and there is no increase in total steady state levels of ubiquitinated proteins in homozygous mutant seedlings. This mutant has seedling growth defects, elevated anthocyanin levels, abnormal leaf root development, delayed skotomorphogenesis, altered cytokinin responses, and reduced auxin sensitivity in their roo

1740:AT1G76030	vha-B1	Insensitive to high-glucose repression of cotyledon expansion, chlorophyll accumulation, true-leaf development and root elongation. The response is specific to glucose but not to osmotic changes.	17081979
1747:AT1G76490	hmg1	Inhibition of root elongation, dwarf, sterile, rapid senescence in mature leaves	14871314
1749:AT1G77110	SALK_046393	shorter visible lateral roots at 9 days after germination.  The proximal region of each lateral root in the mutant is significantly swollen and often bent.  Growth and morphology of the primary root of the puchi mutant is normal.	17630277
1795:AT1G78240	tsd2-1	Increased activity of axial meristems, reduced root growth and enhanced de-etiolation.	17461780
1833:AT1G79580	SALK_143526C	Additional columella root cap and lateral root cap cell layers in the mature embryo and postembryonic root development. Root meristem length and cell number is similar to WT. Extra layer of small, stem-cell-like cells, below the columella and lateral 

2016:AT2G01830	wol	Defect in cell division in the root and lower hypocotyl region soon after the embryo torpedo stage.	11114883
2019:AT2G01830	wol	Reduced cell number and exclusive xylem differentiation within the root vasculature.	11114883
2020:AT2G01830	wol	Root elongation markedly reduced in the presence of kinetin.	15053761
2025:AT2G01830	wol-1:wol-2	Root cross-sections are identical to those of wildtype: observed re-appearance of a bipolar xylem strand and phloem tissue on each side.	15053761
2026:AT2G01830	wol-1:wol-2	Root elongation is indistinguisable from that of wildtype.	15053761
2029:AT2G01830	wol-2	Rapidly accumulate anthocyanins due to the loss of root functionality.	15053761
2030:AT2G01830	wol-2	Short root phenotype.	15053761
2032:AT2G01830	wol-2	Viability depends on the ability to develop adventitious root before plant dies.	15053761
2036:AT2G01830	wol-3	Aborted vascular system containing few protoxylem cells in the primary root. Normal adventitious root-vascular system

2385:AT2G25490	ebf1-1 ebf2-1	The mutant shows dwarfed growth, supernumerary epinastically curled leaves, early senescence, and abnormal flowers with the gynoecium protruding from the unopened floral buds. However, the phenotypes seen in double mutant plants are more severe than those seen in ctr1. This becomes apparent at about two weeks after germination, except for the reduced root growth, which is already obvious at three days after germination. Moreover, unlike the ctr1 mutation that has a reduced transmission of the ctr1 allele relative to wild-type and infertile early flowers, the double mutants have a greatly reduced fertility but can readily be pollinated with wild-type pollen.	14675533
2387:AT2G25490	ebf1-1 ebf2-1	When grown in the light, the double mutants display slightly longer hypocotyls, curled unexpanded cotyledons, and short and thicker roots with ectopic root hairs, similar to the ctr1 mutant. Moreover, the average area of the leaf epidermal cells was reduced 5-fold co

2790:AT2G38120	aux1-115	Root agravitropic phenotype.	15486104
2791:AT2G38120	aux1-116	Partial root gravitropic phenotype.	15486104
2792:AT2G38120	aux1-117	Root agravitropic phenotype.	15486104
2793:AT2G38120	aux1-118	Root agravitropic phenotype.	15486104
2794:AT2G38120	aux1-120	Root agravitropic phenotype.	15486104
2796:AT2G38120	aux1-21	Root agravitropic phenotype.	15486104
2797:AT2G38120	aux1-22	Root agravitropic phenotype.	15486104
2798:AT2G38120	axr4-1 aux1-7	auxin resistant; narrow, irregular rosette leaves, slightly curled around leaf axis; roots elongate on auxin-containing medium; defective root gravitropism; reduced number of lateral roots - greater reduction than for either single mutant; dwarf, bushy plants; reduced plant height; ethylene resistant; reduced fertility.	7704045
2799:AT2G38120	wav5-33	Root agravitropic phenotype.	15486104
2818:AT2G38470	SALK_006603	Homozygotes have a WT phenotype. Root bending and root elongation assays under osmotic and ABA treatments showed n

3094:AT2G46370	CS8072	JA-Ile levels are seven-fold lower in the jar1-1 mutant compared to wild type plants. The levels of JA, JA-Val, and JA-Leu are similar in the wild-type and mutant backgrounds, whereas the level of JA-Phe and JA-ACC are higher in the mutant. Although this mutant is moderately insensitive to MeJA in root growth assays, it is sensitive to JA-Ile.	15258265
3095:AT2G46370	CS8072	The jar1-1 mutant shows moderate insensitivity to MeJA in root growth  inhibition assays, but it has normal male fertility.
3099:AT2G46370	jar1-11	The mutant has reduced sensitivity to methyl jasmonate in root growth inhibition experiements. It also produces lower levels of jasmonyl-isoleucine (JA-Ile) than wild-type plants in response to wounding.	18247047
3101:AT2G46370	jar1-8	The jar1-8 mutant shows moderate insensitivity to MeJA in root growth  inhibition assays.
3127:AT2G47000	mdr4-1	Although mutant seedlings display normal acropetal auxin transport in the root, basipetal auxin transport i

3356:AT3G08040	CS6585	defective in iron translocation; chlorotic plants; expression of iron deficiency responses under conditions of iron sufficiency; overaccumulation of iron and other metals (plants accumulate approximately two-fold excess iron, four-fold excess manganesse and two-fold excess zinc in their shoots); detached roots are capable of repressing iron uptake responses when cultured under iron-sufficient conditions; when shoots are appropriately supplied with iron, they regreen and the plants are capable of correctly regulating root iron uptake responses; isolated protoplasts have lower iron levels than those from wild type plants; plants accumulate abnormal levels of ferric iron in their root vasculature, FRD3 is expressed in the pericycle and other vascular cylinder cells in the mature portion of the root; lacks trichomes on stems and leaves.	12172022
3359:AT3G08710	trx h9	Roots and leaves of  the homozygous mutants were significantly shorter and smaller  than wild-type. Ro

3596:AT3G18780	act2-1	Mutants have abnormal root hairs that are stunted, bulging or may be branched. The orientation of microtubules is abnormal.	19304937
3597:AT3G18780	act2-1/act7-4	Plants are dwarf and reach only 15% of the height of wild type. Roots lack root hairs- the initiate but do not develop further than a bulge. Leaves are small with reduced surface area and fewer lobes. Leaf trichomes are abnormal and contain fewer branches. Also defects in flower morphology, reduced fertility, abnormal inflorescence structure and abnormal silique development.	19304937
3598:AT3G18780	act2-1/act8-2	Lacks root hairs.Root hairs initiate but do not develop further.	19304937
3606:AT3G19050	pok1-1/pok2-1	At seedling stage pok1/pok2 double mutants differ from wild-type by having smaller cotyledons as well as shorter, wider roots and hypocotyls. Adult plants exhibit a dwarfed stature, but all organs are present, although miniaturized. Plants are fertile but produce reduced numbers of seeds. Microsc

3900:AT3G30775	pdh1-2	Decreased root elongation and decreased dry weight for plants grown under low water potential conditions.	21791601
3904:AT3G42170	daysleeper	Seedlings homozygous for the T-DNA insertion completely lacked expression of DAYSLEEPER. Mutant seedlings grew very slowly, and showed no expansion of the cotyledons or development of normal leaves or floral organs. The root remained short and had an excess of abnormal root hairs. The main and lateral root meristems appeared to have a normal organization when young, but as they matured became increasingly disorganized. After prolonged growth on plates, a few seedlings (1%) produced some photosynthetic tissue.	16015335
3910:AT3G43190	sus1 sus4	The double mutant line was much more  severely affected by flooding. After the 4 days of flooding, root weight had increased by only 44% (compared to 72% for wild type), and there was not statistically significant gain in root weight in the 3 days following flooding. At the end of the 7-

4274:AT3G59420	acr4-2	Increased number of lateral root meristems initate.	18948541
4275:AT3G59420	acr4-2/ccr1-3	Increased number of lateral root meristems, greater than acr4 single mutant alone.	18948541
4305:AT3G61650	TUBG1-1/tubg1-1 TUBG2-2/tubg2-2	During early seedling growth, cotyledons were small and distorted, and only two leaf primordia were inittaited at the shoot apex. Dome-shape soot apical meristem was distorted. Root growth was inhibited, and root apex was swollen.	16698945
4313:AT3G61890	athb12-1	Decreased sensitivity to ABA in a root elongation assay with respect to wild type.  The response was consistently approximately 10% less than the wild-type response.	15604708
4328:AT3G62980	CS3798	resistant to auxin inhibition of root elongation; deficient in auxin-regulated growth processes including: reduced lateral root formation, reduced temperature-induced hypocotyl elongation, and a modest reduction in apical dominance.	9436980
4339:AT3G63010	gid1a-1 gid1b-1	Only slight diff

4659:AT4G16760	acx1-3	Plants display decreased sensitivity to the inhibitory effect of indole-3-butyric acid (IBA) on root elongation, while remaining sensitive to inhibitory concentrations of indole-3-acetic acid.  They maintain their ability to initiate lateral roots in response to IBA.	15743450
4712:AT4G18780	CS18	With the exception of the collapsed xylem phenotype, both the organization of the vascular tissue and the organization of the entire stem of the three irx mutants are identical to that of the wild type. In addition to stems, the collapsed xylem phenotype of irx plants has been observed in mature hypocotyls and in the primary root and petioles.	9165747
4826:AT4G24400	cipk8-1	Increased primary root length in the cipk8 mutants. Expression of AttDT, BOR1, CHL1 and NRT2.1, NIA1, NIA2 and NiR was reduced in the mutant.	18798873
4851:AT4G24670	SALK_021258C	This mutant does not have any obvious morphological defects and it responds normally to ACC and IAA in hypocotyl and root elo

5008:AT4G32040	SALK_000339	No visible root phenotype.	16463096
5025:AT4G32700	SALK_018851	Similar to teb-1, with a typical short-root phenotype and the aerial system that is reduced in size, with highly serrated and asymmetric leaves and a fasciated stem.	16517762
5026:AT4G32700	SALK_035610	Similar to teb-1, with a typical short-root phenotype and the aerial system that is reduced in size, with highly serrated and asymmetric leaves and a fasciated stem.	16517762
5028:AT4G32700	teb-1	A short-root phenotype with the phenotype in the aerial part that is similar to those of the tsk mutant. The roots in mutant seedlings were approximately one-third as long as those from wild type plants. This reduced growth became more pronounced at later time points. The root apical meristem was disorganized, and individual cell types, such as QC or initial cells, were difficult to discern. The aerial part of teb-1 plants also showed defects in morphology, such as abnormal phyllotaxy, highly serrated and a

5451:AT5G03280	seo2	Compared with the parent ein2 and EIN3ox plants, etiolated EIN3ox ein2 seedlings exhibited an intermediate phenotype in terms of hypocotyl and root lengths. A reduced EIN3 protein levels were detected in EIN3ox ein2-5 progeny compared with that of EIN3ox parent plants. The EIN3ox ein2 and five seo seedlings were completely unresponsive to ethylene application, similar to ein2. In addition, treatment with ACC failed to enhance the accumulation of EIN3 protein in EIN3ox ein2-5. The expression levels of ERF1 and EBF2 were not further increased by ethylene in the EIN3ox ein2 plant. By contrast, ethylene application enhanced EIN3 protein accumulation and ERF1 and EBF2 gene expression in the EIN3ox plants.	20647342
5452:AT5G03540	exo70A1-1	Early seedling growth was not affected in mutants. Later, mutant seedlings had shorter roots, and retarded lateral rot intiation. The polar growth of root hairs and stigmatic papillae is disturbed. Organs are generally smaller, plants s

5617:AT5G09810	act7-4	Abnormal roots. Roots are shorted and have fewer root hairs (ca. 20% less than wild type). Root epidermal cells have abnormal morphology and bulge.	19304937
5618:AT5G09810	act8-2/act7-4	Plants are dwarf. Leaves are smaller and contain fewer cells and fewer lobes than wild type. Reduced number of trichome branches. Petals are narrow. Roots are more severely affected and produce short root hairs. Cell files are irregular and the elongation zone is much reduced.	19304937
5667:AT5G10250	dot3-1	dot3-1 mutants are dwarfed plants that have a parallel venation pattern defect in their leaves and abnormal vein loop patterns in their cotyledons. Primary root growth is significantly inhibited in dot3-1 mutants, but, they can produce lateral and adventitious roots.	18643975
5668:AT5G10330	hpa1	Short roots. Initial development is normal but by 7 days post germination organization of the root apex begins to show abnormalities.	16815950
5672:AT5G10510	plt3-1	Short roots and short

5972:AT5G20490	SALK_067972C	The xik-2 mutant has normal root growth, but its root hairs are only 50% of the length of wild type root hairs, and they have a hooked shape when grown on media with sugar. The stem trichomes are crooked and wavy and often have abnormally elongated stalks. The trichomes on the leaves also have size and shape irregularities.  But, the total number of trichomes is the same for mutants and wild-type plants.	17458634
5973:AT5G20490	SALK_067972C	There are no obvious defects in the aerial tissues of xi-k mutant plants grown under normal conditions. The root length of the mutant is also normal, but, it has shorter roots hairs than wild-type plants. There are decreased rates of movement of Golgi stacks, peroxisomes, and mitochondria in root hairs and leaves.	18178669
5974:AT5G20490	SALK_136682C	The xik-1 mutant has normal root growth, but its root hairs are only 49% of the length of wild type root hairs, and they have a hooked shape when grown on media with sugar. T

6164:AT5G24270	sos3-1 siz1-1	Decrease of macronutrients concentration in the MS salt formulation to 1/20X results in a substantial reduction in primary root growth.	15894620
6165:AT5G24270	sos3-1 siz1-1	Mutant grown in greenhouse or growth chamber exhibits reduced shoot and root biomass relative to the wild-type.	15894620
6167:AT5G24270	sos3-1 siz1-1	Substantially more pronounced prototypical Pi starvation root architecture responses than sos3-1 or wild-type seedlings. On Pi-limited medium, mutant seedlings exhibit an inhibition of primary root growth, an increase of lateral root development and length, and higher root/shoot fresh weight ratio than wild-type.  Enhanced resistance to Pseudomonas syringae DC3000.  Smaller in stature.  Impaired drought tolerance.	15894620
6169:AT5G24310	amiRNA_ABIL3	Trichomes are irregularly expanded and distorted. Branch  lengths are significantly affected with all  three branches being significantly shorter than the respective branches of wild type tric

6362:AT5G35750	stip-d ahk2-2 ahk3-3 cre1-12	Small SAM and limited root growth.	20110319
6378:AT5G37830	SALK_078745C	oxp1-1 cannot convert 5-oxoproline (5OP) into glutamine. Elevated levels of 5OP accumulate in the the leaves, stems, roots, flowers, and siliques of this mutant, while the level of glutamate is reduced. oxp1-1 mutants have normal seedling root growth and do not display any other morphological defects. They appear to have normal levels of Asp, Ser, Gln, Gly, Thr, Arg, and Ala, and have a wild-type sensitivity to nitrogen deficiency.	18768907
6389:AT5G39510	sgr4-1	Inflorescence stems and hypocotyls of mutant plants showed abnormal gravitropic response, while root gravitropism was normal.	9210330
6394:AT5G39610	nac2-1	Not significantly affected in lateral root development.	16359384
6440:AT5G42890	Atscp2-1	Lower germination frequency and time, restored by addition of 1% sucrose. Smaller cotyledon rosettes and  shorter hypocotyls than wild-type and the root elongation  was als

6629:AT5G49680	kip-2/sab	Adult mutant plants were three to four times smaller than single mutant, sab plants. Roots of double mutant were shorter than roots of sab mutant. Radial organization of the root was similar to the roots of sab mutant. Root hairs were thicker that in wild type or sab mutant. Pollen grains from double mutant plants rarely germinated in vitro, and when they did, the pollen tubes were shorter and twisted.	14675453
6630:AT5G49680	kip-3	Pollen tube growth was altered. Pollen tubes were shorter, often having branched appearance and peculiar kinky shape. Also, callose plugs were rare or absent in mutant pollen tubes, compared to the wild type. Despite morphological aberrations, twisted pollen tube appeared to have same ultrastructural features as the wild type pollen tubes. Migration of the male germ unit to the pollen tube occurred at approximately the same time in mutants and in the wild type. Homozygous progeny showed shorter and thicker root hairs. No other signif

6931:AT5G60410	sos3-1 siz1-1	Decrease of macronutrients concentration in the MS salt formulation to 1/20X results in a substantial reduction in primary root growth.	15894620
6932:AT5G60410	sos3-1 siz1-1	Mutant grown in greenhouse or growth chamber exhibits reduced shoot and root biomass relative to the wild-type.	15894620
6934:AT5G60410	sos3-1 siz1-1	Substantially more pronounced prototypical Pi starvation root architecture responses than sos3-1 or wild-type seedlings. On Pi-limited medium, mutant seedlings exhibit an inhibition of primary root growth, an increase of lateral root development and length, and higher root/shoot fresh weight ratio than wild-type.  Enhanced resistance to Pseudomonas syringae DC3000.  Smaller in stature.  Impaired drought tolerance.	15894620
6947:AT5G60920	CS8541	When grown on media containing 0.5% sucrose, the mutant had reduced root length but no apparent radial cell expansion.	7743935
6948:AT5G60920	cob-4	Dark grown hypocotyls have reduced length compared

Binary file Locus_Germplasm_Phenotype_20130122.txt matches


# Problem 7

In the previous excercise I already selected the lines with "root" on their phenotype, so following those steps I save the output on a file named ABBA. Then, made a *grep* command with the *-E* flag for regular expressions and a *-io* to avoid case specificness and to show only the word required by the *grep*. In this case the AGI Locus code is formed by two possile letters (A-a) or (T-t) in the first two possitions, so [A-Z] means the must be letters and {2} specifies the two first positions. Then it comes the chromosome number, from [1-5] and another letter [A-Z]. The last 5 digits are numbers from [0-9] and the {5} is specified. The output of this *grep* is stored in a new file named Root-associated-LOci.txt. To make sure all AGI codes were taken, a *wc* command was used. Apparently the ABBA file is slightly bigger than the Root-associated... this is because some of the last ABBA lines do not show a AGI code, but a 4 character identificator wich could mean they do not have the AGI code or any other reason. Those unknown codes were not asked for and thus not extracted. A *cat* command is done to make sure the file created is ok, and after that both are removed so the next time this script runs they wont be overwritten.

In [76]:
grep -inw "root" Locus_Germplasm_Phenotype_20130122.txt >> ABBA
grep -E -io "\w{2}[1-5][A-Z][0-9]{5}" ABBA >> Root-associated-Loci.txt
wc -l Root-associated-Loci.txt
wc -l ABBA
cat Root-associated-Loci.txt
rm ABBA
rm Root-associated-Loci.txt

617 Root-associated-Loci.txt
628 ABBA
AT1G01550
AT1G01550
AT1G01550
AT1G02970
AT1G03000
AT1G03060
AT1G03060
AT1G03060
AT1G03060
AT1G04020
AT1G04030
AT1G04160
AT1G04160
AT1G04160
AT1G04220
AT1G04250
AT1G05180
AT1G05180
AT1G05850
AT1G05850
AT1G06290
AT1G06290
AT1G08030
AT1G08260
AT1G08370
AT1G08430
AT1G09100
AT1G09110
AT1G10130
AT1G10570
AT1G11130
AT1G15550
AT1G15690
AT1G16150
AT1G17110
AT1G17170
AT1G17580
AT1G17580
AT1G18800
AT1G19220
AT1G19220
AT1G19220
AT1G19220
AT1G19220
AT1G19220
AT1G19220
AT1G19220
AT1G19220
AT1G21700
AT1G21970
AT1G21970
AT1G22920
AT1G22920
AT1G23010
AT1G23010
AT1G23310
AT1G23320
AT1G23320
AT1G26870
AT1G27320
AT1G27320
AT1G27320
AT1G27320
AT1G27320
AT1G27320
AT1G27320
AT1G27320
AT1G27320
AT1G27320
AT1G27320
AT1G27320
AT1G27320
AT1G27320
AT1G27320
AT1G27760
AT1G28560
AT1G29990
AT1G31480
AT1G31880
AT1G31880
AT1G31880
AT1G31880
AT1G32450
AT1G33280
AT1G33280
AT1G33290
AT1G33290
AT1G34370
AT1G34370
AT1G34370
AT1G35580
AT1G47056
AT1G47720
AT1G47720
AT1G48380
AT1G48920
AT

# Problem 8

Similarly to the previous excercise, the file containig only the lines with "root" on their phenotypes is named ABBA. Its number of lines are displayed for further comprobations with the *wc -l* command. To extract the pubmed id of every line we follow a similar strategy as in the excercise avobe and start a *grep* command searching for space "\s", numbers from 0 to 9 in a sequence "+" (more than one) and at the end of the line "$" and the *-ino* flags. Theoutput is stored in a file named Root-associated-Publication.txt, and the number of lines displayed by *wc*. But the lines inside Root-associated... was way smaller than ABBA file (558 vs 628). In order to assure that all the pubmed ids were correctly extracted I did a *cat* command with the *-n* flag to see the line number, so whenever I found a disonance between the ABBA line number and the Root... line number (remember that the *grep* that extracted the ABBA from the Locus_Germplasm file had the -n flag, and theorically all lines containing a phenotype should have a pubmed id, thus the file should had same number of lines and each line the same number). 
The first disonance is on line 26 of the ABBA file, which was not *grep*-ed to the Root... file. Searching for it on the ABBA file, I discovered that line (line 257 on the Locus_Germplasm... file) has no pubmed identificator. This can be a mistake during the development of the file, or simply there is no pubmed file. Assuming that my code was correct, and that the missing lines are only due to the lack of associated publications I finally removed the created files (*rm*)



In [121]:
grep -inw "root" Locus_Germplasm_Phenotype_20130122.txt >> ABBA
wc -l ABBA
grep -E -ino "\s[0-9]+$" ABBA>> Root-associated-Publications.txt
wc -l Root-associated-Publications.txt
cat -n Root-associated-Publications.txt
rm ABBA
rm Root-associated-Publications.txt

628 ABBA
558 Root-associated-Publications.txt
     1	1:	17217459
     2	2:	15458645
     3	3:	17217459
     4	4:	17209125
     5	5:	14745029
     6	6:	19392685
     7	7:	19392685
     8	8:	19392685
     9	9:	19392685
    10	10:	18591352
    11	11:	18591352
    12	12:	18178669
    13	13:	18178669
    14	14:	19060218
    15	15:	18786002
    16	16:	16367964
    17	17:	17655650
    18	18:	7704045
    19	19:	7743935
    20	20:	7743935
    21	21:	15743450
    22	22:	15743450
    23	23:	19666544
    24	24:	16278345
    25	25:	17485080
    26	27:	17081979
    27	28:	17081979
    28	29:	18024560
    29	30:	18849491
    30	31:	15618487
    31	32:	16460513
    32	33:	16210544
    33	34:	16286448
    34	35:	18485060
    35	37:	18178669
    36	38:	19060218
    37	40:	15659631
    38	41:	15659631
    39	42:	15659631
    40	43:	15659631
    41	44:	15659631
    42	45:	15659631
    43	46:	15659631
    44	47:	15659631
    45	48:	16461383
    46	50:	12244233
    47	51:	12244233
    48	52:	17307927
    49

   395	447:	16766694
   396	449:	17785527
   397	450:	17785527
   398	451:	17533512
   399	453:	20647342
   400	454:	16942608
   401	455:	16942608
   402	456:	8431946
   403	457:	8431946
   404	458:	8431946
   405	459:	8431946
   406	460:	8431946
   407	461:	15272001
   408	462:	15272001
   409	463:	15978034
   410	464:	21800258
   411	465:	21800258
   412	466:	21800258
   413	467:	21800258
   414	468:	17012601
   415	469:	17012601
   416	470:	20230485
   417	471:	19304937
   418	472:	19304937
   419	473:	19304937
   420	474:	18643975
   421	475:	16815950
   422	476:	17960244
   423	477:	17960244
   424	478:	9367981
   425	479:	18065688
   426	480:	9367981
   427	481:	9367981
   428	482:	9367981
   429	484:	17158604
   430	485:	17485080
   431	486:	17513503
   432	487:	17485080
   433	488:	17513503
   434	489:	17586653
   435	490:	19506555
   436	491:	8643469
   437	492:	17630277
   438	493:	17062755
   439	494:	17363254
   440	495:	17363254
   441	496:	17662035
   442	497:	17458634
  

# Problem 9

The AGI locus code states that the third caracter (number from 1-5) refers to the number of chromosome the gene is located in. THus, if the hypothesis is correct we should find in root associated genes' AGIs examples of all numbers from 1 to 5 at least once. To demonstrate this and as done in pervious excercises I isolated the "root" cointaining phenotypes and stored them in a file named ABBA. After that I isolated the AGI codes just like in excercise 7 on another file named Root-associated-Loci.txt. By doing several *grep* on that file looking for the specificity on the third character (*\w{2}* first two leters don't matter, third charecter ranging from 1 to 5 and another unimportant 6 word characters *\w{6}*) and storing each possible output in a different file (Chr 1 to 5, meaning chromosomes 1 to 5), I can then classify the AGIs of the genes according to the chromosome they are found on. If all of those new archives have at least one AGI this would mean that our hypothesis is correct, this can be proved by measuring the number of lines in each file with a simple *wc -l* command. After demonstrating that indeed genes assocaited with roots can be found on all chromosomes, I removed all files to avoid overwritting.

In [7]:
grep -inw "root" Locus_Germplasm_Phenotype_20130122.txt >> ABBA
grep -E -io "\w{2}[1-5][A-Z][0-9]{5}" ABBA >> Root-associated-Loci.txt
wc -l Root-associated-Loci.txt
grep -E -io "\w{2}1\w{6}" Root-associated-Loci.txt >> Chr1
grep -E -io "\w{2}2\w{6}" Root-associated-Loci.txt>> Chr2
grep -E -io "\w{2}3\w{6}" Root-associated-Loci.txt>> Chr3
grep -E -io "\w{2}4\w{6}" Root-associated-Loci.txt>> Chr4
grep -E -io "\w{2}5\w{6}" Root-associated-Loci.txt>> Chr5
wc -l Chr1
wc -l Chr2
wc -l Chr3
wc -l Chr4
wc -l Chr5
rm Root-associated-Loci.txt
rm ABBA
rm Chr1
rm Chr2
rm Chr3
rm Chr4
rm Chr5

617 Root-associated-Loci.txt
159 Chr1
134 Chr2
96 Chr3
56 Chr4
172 Chr5
